pkg/azuredisk/controllerserver.go

```diff
@@ -879,13 +885,28 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		snapshot.SnapshotProperties.DataAccessAuthMode = compute.DataAccessAuthMode(dataAccessAuthMode)
 	}
 
+	var crossRegionSnapshotName string
+	if location != "" && location != d.cloud.Location {
+		if incremental {
+			crossRegionSnapshotName = snapshotName
+			snapshotName = azureutils.CreateValidDiskName("local_" + snapshotName)
+		} else {
+			return nil, status.Errorf(codes.InvalidArgument, "could not create snapshot cross region with incremental is false")
+		}
+	}
+
+	if acquired := d.volumeLocks.TryAcquire(snapshotName); !acquired {
```

Comment:
move to L887, let's only lock on target snapshotName

Reply:
done